File: 1. Two Sum/1.cc
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        vector<pair<int, int>> sorted;

        for (int i = 0; i < n; ++i) {
            sorted.push_back(make_pair(nums[i], i));
        }
        sort(sorted.begin(), sorted.end(), less<>());

        int left = 0;
        int right = n - 1;
        while (left < right) {
            int sum = sorted[left].first + sorted[right].first;
            if (sum < target) {
                ++left;
            } else if (sum > target) {
                --right;
            } else {
                return {sorted[left].second, sorted[right].second};
            }
        }
        return {};
    }
};

File: 1. Two Sum/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sorted = []
        for i, x in enumerate(nums):
            sorted.append([x, i])
        sorted.sort()

        left, right = 0, len(nums) - 1
        while left < right:
            sum = sorted[left][0] + sorted[right][0]
            if sum < target:
                left += 1
            elif sum > target:
                right -= 1
            else:
                return [sorted[left][1], sorted[right][1]]
        return []
